File: backend/src/main/java/com/happy/friendogly/pet/domain/Gender.java
package com.happy.friendogly.pet.domain;

import com.happy.friendogly.exception.FriendoglyException;
import java.util.Set;
import java.util.stream.Collectors;

public enum Gender {

    MALE,
    FEMALE,
    MALE_NEUTERED,
    FEMALE_NEUTERED;

    public static Gender toGender(String gender) {
        try {
            return valueOf(gender);
        } catch (IllegalArgumentException e) {
            throw new FriendoglyException("존재하지 않는 Gender 입니다.");
        }
    }

    public static Set<Gender> toGenders(Set<String> genders) {
        return genders.stream()
                .map(Gender::toGender)
                .collect(Collectors.toSet());
    }
}

File: android/app/src/main/java/com/happy/friendogly/domain/model/Gender.kt
package com.happy.friendogly.domain.model

enum class Gender(val petGender: String) {
    MALE("수컷"),
    FEMALE("암컷"),
    MALE_NEUTERED("중성화 수컷"),
    FEMALE_NEUTERED("중성화 암컷"),
}
